applications/nrf_desktop/src/modules/led_stream.c

```diff
@@ -0,0 +1,218 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <zephyr.h>
+#include <misc/byteorder.h>
+
+#define MODULE led_stream
+#include "module_state_event.h"
+
+#include <logging/log.h>
+
+#include "config_event.h"
+#include "led_event.h"
+
+#define STREAM_LED_ID 0
+#define INCOMING_LED_COLOR_COUNT 3
+#define STREAM_DATA_SIZE 7
+
+LOG_MODULE_REGISTER(MODULE, CONFIG_DESKTOP_LED_STREAM_LOG_LEVEL);
+
+static struct led_effect led_stream_effect = {
+	.step_count = 1,
+};
+static struct led_effect_step steps_queue[CONFIG_DESKTOP_LED_STREAM_QUEUE_SIZE];
+static u8_t rx_idx;
+static u8_t tx_idx;
+static bool first_sent;
+static bool leds_initialized;
+
+static void queue_data(const struct event_dyndata *dyndata)
+{
+	LOG_INF("Insert data to effect");
+	size_t pos = 0;
+
+	memcpy(steps_queue[rx_idx].color.c, &dyndata->data[pos],
+	       CONFIG_DESKTOP_LED_COLOR_COUNT);
+	pos += INCOMING_LED_COLOR_COUNT;
+	steps_queue[rx_idx].substep_count = sys_get_le16(&dyndata->data[pos]);
+	pos += sizeof(steps_queue[rx_idx].substep_count);
+	steps_queue[rx_idx].substep_time = sys_get_le16(&dyndata->data[pos]);
+
+	rx_idx = (rx_idx + 1) % CONFIG_DESKTOP_LED_STREAM_QUEUE_SIZE;
```

Comment:
I suggest creating a small function
```
static size_t next_index(size_t index)
{
   return (index + 1) % CONFIG_DESKTOP_LED_STREAM_QUEUE_SIZE
}
```

as I expect this is executed in many places

Reply:
done